clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -2175,7 +2182,7 @@ public void testClusterAuthorizationExceptionInProduceRequest() throws Exception
         sender.runOnce();
         assertFutureFailure(future, ClusterAuthorizationException.class);
 
-        // cluster authorization errors are fatal, so we should continue seeing it on future sends
+        // expecting to continue to see authorization error until user permission is fixed
         assertTrue(transactionManager.hasFatalError());
```

Comment:
Should this still be true? Do we remove it from fatal error state?

Reply:
I think that's actually my mistake.  It's a produce error not initProducerId.  I'll correct these issues. Sorry.

Reply:
Ah I also missed. Thanks for correcting